### PR TITLE
Updated src/OpenTelemetry and test/OpenTelemetry.Tests files to use f…

### LINKED ISSUE #4640: Use file scoped namespaces
Update the code files in the repo to use file scoped namespaces: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces

This should be done in smaller PRs each preferably targeting three to four `src` projects and their corresponding `test` projects in one PR.

### PR DESCRIPTION
**Pull request by kyle-lt**
…ile scoped namespaces

This is the first of multiple pull requests, first time around for just the core OpenTelemetry src and tests projects, and if things look good, I'll batch up more at a time in the ensuing PRs.

Partially Fixes #4640

## Changes

Simple formatting updates to use file scoped namespaces.

## Merge requirement checklist

* [ X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X ] Unit tests added/updated
* [X ] Appropriate `CHANGELOG.md` files updated for non-trivial changes - trivial style/formatting change
* [ X] Changes in public API reviewed (if applicable) - N/A

### CODE REVIEW

**Review by reyang — APPROVED** [review]

**Comment by reyang** [comment]
Thanks @kyle-lt!

File: .editorconfig
Comment (vishweshbankwar) [inline .editorconfig:1]: @kyle-lt Why is this file deleted?

Comment (kyle-lt) [reply .editorconfig:1]: @vishweshbankwar That was a mistake, and so I reverted it.  Thanks for calling it out!

**Comment by utpilla** [comment]
@kyle-lt Thanks for sending the PR! It looks good to be merged. Please resolve the merge conflicts.

@open-telemetry/dotnet-approvers Check the Hide Whitespace option in the Diff View to make it easy for review.

**Comment by kyle-lt** [comment]
Is there a way to rerun the failed test?  There was a timeout fetching a resource, according to the [logs](https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/5548243237/jobs/10130971689?pr=4654#step:3:320).

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
> Is there a way to rerun the failed test? There was a timeout fetching a resource, according to the [logs](https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/5548243237/jobs/10130971689?pr=4654#step:3:320).

The maintainers have permissions to re-run the tests.

For now, all of the CI checks have passed.
